src/DatadirTestCase.php

```diff
@@ -0,0 +1,194 @@
+
+    public function __construct(
+        ?string $name = null,
+        array $data = [],
+        string $dataName = ''
+    ) {
+        $reflectionClass = new ReflectionClass(static::class);
+        $this->testFileDir = dirname($reflectionClass->getFileName());
+        parent::__construct($name, $data, $dataName);
+    }
+
+    /**
+     * @dataProvider provideDatadirSpecifications
+     */
+    public function testDatadir(DatadirTestSpecificationInterface $specification): void
+    {
+        $fs = new Filesystem();
+
+        $script = $this->getScript();
+        if (!$fs->exists($script)) {
+            throw new Exception(sprintf(
+                'Cannot open script file "%s"',
+                $script
+            ));
+        }
+
+        $temp = new Temp();
+        $temp->initRunFolder();
+
+        $fs->mirror($specification->getWorkingDirectory() . '/source/data', $temp->getTmpFolder());
+
+        $fs->mkdir($temp->getTmpFolder() . '/in/tables', 0777);
+        $fs->mkdir($temp->getTmpFolder() . '/in/files', 0777);
+        $fs->mkdir($temp->getTmpFolder() . '/out/tables', 0777);
+        $fs->mkdir($temp->getTmpFolder() . '/out/files', 0777);
+
+        $runCommand = [
+            "php",
+            $script,
+        ];
+        $runProcess = new Process($runCommand);
+        $runProcess->setEnv([
+            'KBC_DATADIR' => $temp->getTmpFolder(),
+        ]);
+        $runProcess->run();
+
+        if ($specification->getExpectedReturnCode() !== null) {
+            $this->assertProcessReturnCode($specification->getExpectedReturnCode(), $runProcess);
+        } else {
+            $this->assertNotSame(0, $runProcess->getExitCode(), 'Exit code should have been non-zero');
+        }
+        if ($specification->getExpectedStdout() !== null) {
+            $this->assertSame(
+                $specification->getExpectedStdout(),
+                $runProcess->getOutput(),
+                'Failed asserting stdout output'
+            );
+        }
+        if ($specification->getExpectedStderr() !== null) {
+            $this->assertSame(
+                $specification->getExpectedStderr(),
+                $runProcess->getErrorOutput(),
+                'Failed asserting stderr output'
+            );
+        }
+        if ($specification->getExpectedOutputDirectory() !== null) {
+            $this->assertDirectoryContentsSame(
+                $specification->getWorkingDirectory() . '/expected/data/out',
+                $temp->getTmpFolder() . '/out'
+            );
+        }
+    }
+
+    /**
+     * @return DatadirTestSpecificationInterface[][]
+     */
+    public function provideDatadirSpecifications(): array
+    {
+        $data = [[]];
+        foreach ($this->getDataProviders() as $provider) {
+            $data[] = $provider();
+        }
+        // https://github.com/kalessil/phpinspectionsea/blob/master/docs/performance.md#slow-array-function-used-in-loop
+        return array_merge(...$data);
```

Comment:
to je super. i z pohledu srozumitelnosti mi to přijde mnohem lepší než ten loop.